### PR TITLE
adding handle of missing conversation

### PR DESCRIPTION
**Pull request by elopezanaya**
For GetPostChatSurveyContext , when liveworkitem details is returninig null, it wont continue with the flow.

image

image

## ENDED by Agent

image

### Ended by customer
- not showing survey

### CODE REVIEW

File: src/OmnichannelChatSDK.ts [hunk 1]
```diff
@@ -1707,11 +1708,22 @@ class OmnichannelChatSDK {
             const { msdyn_postconversationsurveyenable, msfp_sourcesurveyidentifier, msfp_botsourcesurveyidentifier, postConversationSurveyOwnerId, postConversationBotSurveyOwnerId } = liveWSAndLiveChatEngJoin;
 
             if (parseLowerCaseString(msdyn_postconversationsurveyenable) === "true") {
+
                 const liveWorkItemDetails = await this.getConversationDetails();
+                
+                if ( Object.keys(liveWorkItemDetails).length == 0) {
```
Comment (kevinlin98z) [inline src/OmnichannelChatSDK.ts:1714]: nit: spacing

**Review by kevinlin98z — APPROVED** [review]

File: src/OmnichannelChatSDK.ts [hunk 2]
```diff
@@ -2,14 +2,15 @@
...
-import { ChatMessageReceivedEvent, ChatMessageEditedEvent, ParticipantsRemovedEvent } from '@azure/communication-signaling';
+import { ChatMessageEditedEvent, ChatMessageReceivedEvent, ParticipantsRemovedEvent } from '@azure/communication-signaling';
 import { SDKProvider as OCSDKProvider, uuidv4 } from "@microsoft/ocsdk";
 import { createACSAdapter, createDirectLine, createIC3Adapter } from "./utils/chatAdapterCreators";
 import { defaultLocaleId, getLocaleStringFromId } from "./utils/locale";
 import { isClientIdNotFoundErrorMessage, isCustomerMessage } from "./utils/utilities";
 import { loadScript, removeElementById } from "./utils/WebUtils";
 import platform, { isBrowser } from "./utils/platform";
 import validateSDKConfig, { defaultChatSDKConfig } from "./validators/SDKConfigValidators";
+
```
Comment (xTEddie) [inline src/OmnichannelChatSDK.ts:13]: nit: extra line

File: src/OmnichannelChatSDK.ts [hunk 3]
```diff
@@ -1708,10 +1710,19 @@ class OmnichannelChatSDK {
 
             if (parseLowerCaseString(msdyn_postconversationsurveyenable) === "true") {
                 const liveWorkItemDetails = await this.getConversationDetails();
+                if (Object.keys(liveWorkItemDetails).length == 0) {
+                    this.scenarioMarker.failScenario(TelemetryEvent.GetPostChatSurveyContext, {
```
Comment (xTEddie) [inline src/OmnichannelChatSDK.ts:1714]: I'd suggest to add `ChatId` as well

**Comment by xTEddie** [comment]
I'd add "Fix `ChatSDK.getPostConversationContext()` to reject promise when conversation is not found" in the `CHANGELOG.md`

**Review by nareshme2016 — APPROVED** [review]

File: src/OmnichannelChatSDK.ts [hunk 4]
```diff
@@ -316,9 +317,8 @@ class OmnichannelChatSDK {
 
         return context;
     }
-
+
```
Comment (xTEddie) [inline src/OmnichannelChatSDK.ts:320]: nit: extra space

File: src/OmnichannelChatSDK.ts [hunk 5]
```diff
@@ -1708,10 +1709,20 @@ class OmnichannelChatSDK {
 
             if (parseLowerCaseString(msdyn_postconversationsurveyenable) === "true") {
                 const liveWorkItemDetails = await this.getConversationDetails();
+                if (Object.keys(liveWorkItemDetails).length == 0) {
```
Comment (xTEddie) [inline src/OmnichannelChatSDK.ts:1712]: Use `===`

**Review by xTEddie — APPROVED** [review]

**Pull request merged**